core/src/main/java/hudson/model/Run.java

```diff
@@ -659,20 +679,42 @@ public final long getStartTimeInMillis() {
         return startTime;
     }
 
+    //TODO(oleg_nenashev): Do we want to export full description by default taking the size considerations? Looks like it should be another REST API or a flag
     @Exported
+    @CheckForNull
     public String getDescription() {
         return description;
     }
 
+    /**
+     * Gets summary of the run.
+     * This is a formatted string, {@link hudson.markup.MarkupFormatter} should be applied to it in the WebUI.
+     * If the {@link #summary} field is {@code null}, the method will also try to retrieve a truncated {@link #description}.
+     * @since TODO
+     */
+    @Exported
+    @CheckForNull
+    public String getSummary() {
+        return summary != null ? summary : getTruncatedDescription();
+    }
 
     /**
      * Returns the length-limited description.
+     * The method tries to take the HTML tags into account, but it is a best-effort attempt.
+     * Also, the method will not work properly if a non-HTML {@link hudson.markup.MarkupFormatter} is used.
      * @return The length-limited description.
-     * @deprecated truncated description uses arbitrary and unconfigurable limit of 100 symbols
+     * @deprecated truncated description based on the {@link #TRUNCATED_DESCRIPTION_LIMIT} setting.
      */
     @Deprecated
```

Comment:
should this be un deprecated?

Reply:
I think no, because `getSummary` is a proper API.

Reply:
Yes, I would prefer to not change the state here.

Reply:
I mentioned it as this PR adds documentation for it:
https://github.com/jenkinsci/jenkins/pull/4489/files#diff-c4f9931d88bca347279b881007d71f0eR234

Probably shouldn't be suggesting to call a deprecated API?

Reply:
when compiling, we will have a warning because we are calling a deprecated api. I don't have solution, just worse mentioning that this is intentional, no?